dtfs-central-api/src/helpers/amendments/filter-and-sort-completed-effective-amendments.test.ts

```diff
@@ -0,0 +1,139 @@
+import { addDays, subDays } from 'date-fns';
+import { aCompletedTfmFacilityAmendment, aTfmFacilityAmendment } from '../../../test-helpers';
+import { filterAndSortCompletedEffectiveAmendments } from './filter-and-sort-completed-effective-amendments';
+
+describe('filter-and-sort-completed--effective-amendments', () => {
+  describe('filterAndSortCompletedEffectiveAmendments', () => {
+    it('should filter out all amendments which are not completed', () => {
+      // Arrange
+      const now = new Date();
+      const beforeNow = subDays(now, 1);
+
+      const aCompletedAmendment = { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.COMPLETED, effectiveDate: beforeNow.getTime() };
+      const amendments: TfmFacilityAmendment[] = [
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.IN_PROGRESS, effectiveDate: beforeNow.getTime() },
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.NOT_STARTED, effectiveDate: beforeNow.getTime() },
+        aCompletedAmendment,
+      ];
+
+      // Act
+      const result = filterAndSortCompletedEffectiveAmendments(amendments, now);
+
+      // Assert
+      expect(result).toEqual([aCompletedAmendment]);
+    });
+
+    it('should filter out all amendments with effective date after latestEffectiveDate', () => {
+      // Arrange
+      const latestEffectiveDate = new Date('2024-02-28');
+      const beforeLatestEffectiveDate = subDays(latestEffectiveDate, 1);
+      const afterLatestEffectiveDate = addDays(latestEffectiveDate, 1);
+
+      const amendmentEffectiveBeforeLatestEffectiveDate = {
+        ...aCompletedTfmFacilityAmendment(),
+        effectiveDate: beforeLatestEffectiveDate.getTime(),
+      };
+
+      const amendmentEffectiveAfterLatestEffectiveDate = {
+        ...aCompletedTfmFacilityAmendment(),
+        effectiveDate: afterLatestEffectiveDate.getTime(),
+      };
+
+      const amendments: TfmFacilityAmendment[] = [amendmentEffectiveBeforeLatestEffectiveDate, amendmentEffectiveAfterLatestEffectiveDate];
+
+      // Act
+      const result = filterAndSortCompletedEffectiveAmendments(amendments, latestEffectiveDate);
+
+      // Assert
+      expect(result).toEqual([amendmentEffectiveBeforeLatestEffectiveDate]);
+    });
+
+    it('should filter out amendments with no effective date whose updated at is after the latest effective date', () => {
+      // Arrange
+      const latestEffectiveDate = new Date('2024-02-28');
+      const beforeLatestEffectiveDate = subDays(latestEffectiveDate, 1);
+      const afterLatestEffectiveDate = addDays(latestEffectiveDate, 1);
+
+      const amendmentUpdatedAtBeforeLatestEffectiveDate = {
+        ...aCompletedTfmFacilityAmendment(),
+        effectiveDate: undefined,
+        updatedAt: beforeLatestEffectiveDate.getTime(),
+      };
+
+      const amendmentUpdatedAtAfterLatestEffectiveDate = {
+        ...aCompletedTfmFacilityAmendment(),
+        effectiveDate: undefined,
```

Comment:
Could these be null instead of undefined? Not a blocker

Reply:
No, the type is of the field is of type `number` or `undefined`, here we explicitly set to undefined because we want to ensure it is undefined in the test rather than not specifying it and relying on the `aTfmFacilityAmendment` setting it to undefined and then coupling the success/failure of this test to that function definition